sale_commission_agent_restrict/models/res_partner.py

```diff
@@ -35,3 +35,18 @@ def create(self, vals):
             vals["agent_ids"] = [(4, self.env.user.partner_id.id, 0)]
         res = super(ResPartner, self).create(vals)
         return res
+
+    @api.model
+    def _commercial_fields(self):
+        res = super()._commercial_fields()
+        for agent in self.agent_ids:
```

Comment:
@ilyasProgrammer This is not correct as function is a model decorated one.

You should find another way.

Reply:
AFAIS is not decorated w/ `@api.model` https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/res_partner.py#L405 and is - apparently  - called always on the right recordset.